sklearn/neighbors/lmnn.py

```diff
@@ -0,0 +1,1142 @@
+
+                if len(imp_ind):
+                    # sample impostors if they are too many
+                    if len(imp_ind) > self.max_impostors:
+                        imp_ind = self.random_state_.choice(
+                            imp_ind, self.max_impostors, replace=False)
+
+                    dims = (len(ind_out), len(ind_in))
+                    ii, jj = np.unravel_index(imp_ind, dims=dims)
+                    # Convert indices to refer to the original data matrix
+                    imp_row = ind_out[ii]
+                    imp_col = ind_in[jj]
+                    new_imp = csr_matrix((np.ones(len(imp_row), dtype=np.int8),
+                                          (imp_row, imp_col)), dtype=np.int8,
+                                         shape=(n_samples, n_samples))
+                    impostors_sp = impostors_sp + new_imp
+
+            impostors_sp = impostors_sp.tocoo(copy=False)
+            imp_row = impostors_sp.row
+            imp_col = impostors_sp.col
+
+            # Make sure we do not exceed max_impostors
+            n_impostors = len(imp_row)
+            if n_impostors > self.max_impostors:
+                ind_sampled = self.random_state_.choice(
+                    n_impostors, self.max_impostors, replace=False)
+                imp_row = imp_row[ind_sampled]
+                imp_col = imp_col[ind_sampled]
+
+            imp_dist = _paired_distances_blockwise(X_embedded, imp_row,
+                                                   imp_col)
+        else:
+            # Initialize lists for impostors storage
+            imp_row, imp_col, imp_dist = [], [], []
+            for class_id in classes[:-1]:
+                ind_in, = np.where(y == class_id)
+                ind_out, = np.where(y > class_id)
+
+                # Split ind_out x ind_in into chunks of a size that fits in
+                # memory
+                imp_ind, dist_batch = _find_impostors_blockwise(
+                    X_embedded[ind_out], X_embedded[ind_in],
+                    margin_radii[ind_out], margin_radii[ind_in],
+                    return_distance=True)
+
+                if len(imp_ind):
+                    # sample impostors if they are too many
+                    if len(imp_ind) > self.max_impostors:
+                        ind_sampled = self.random_state_.choice(
+                            len(imp_ind), self.max_impostors, replace=False)
+                        imp_ind = imp_ind[ind_sampled]
+                        dist_batch = dist_batch[ind_sampled]
+
+                    dims = (len(ind_out), len(ind_in))
+                    ii, jj = np.unravel_index(imp_ind, dims=dims)
+                    # Convert indices to refer to the original data matrix
+                    imp_row.extend(ind_out[ii])
+                    imp_col.extend(ind_in[jj])
+                    imp_dist.extend(dist_batch)
+
+            imp_row = np.asarray(imp_row, dtype=np.intp)
```

Comment:
here and one line below: isn't np.int64 more explicit ?

Reply:
Sure, but I think there is still support for 32-bit platforms, no?